src/sinks/util/compressor.rs

```diff
@@ -53,25 +54,28 @@ impl io::Write for Writer {
 ///
 /// Users can acquire a `Compressor` via [`Compressor::from`] based on the desired compression scheme.
 pub struct Compressor {
+    compression: Compression,
     inner: Writer,
 }
 
 impl Compressor {
-    /// Creates a zlib-based compressor with the default compression level.
-    pub fn zlib_default() -> Self {
-        let buf = BytesMut::with_capacity(1_024);
-        Self {
-            inner: Writer::Zlib(ZlibEncoder::new(
-                buf.writer(),
-                flate2::Compression::default(),
-            )),
-        }
-    }
-
```

Comment:
This was here because the normal way to get a `Compressor` is to take the `Compression` enum variant of your choice, and use the `impl From<Compression> for Compressor` conversion to get the compressor... except we never had `Compression::Zlib` due to some previous limitation based on Rusoto.

Since we no longer use Rusoto, I cleaned this up.